youtube_dl/extractor/streamcz.py

```diff
@@ -1,105 +1,112 @@
-class StreamCZIE(InfoExtractor):
-    _VALID_URL = r'https?://(?:www\.)?stream\.cz/.+/(?P<id>[0-9]+)'
-    _API_URL = 'http://www.stream.cz/API'
+        # PLS might contain further formats (DASH, HLS, HLS with fMP4)
+        pls = sdn_data.get('pls')
 
-    _TESTS = [{
-        'url': 'http://www.stream.cz/peklonataliri/765767-ecka-pro-deti',
-        'md5': '934bb6a6d220d99c010783c9719960d5',
-        'info_dict': {
-            'id': '765767',
-            'ext': 'mp4',
-            'title': 'Peklo na talíři: Éčka pro děti',
-            'description': 'Taška s grónskou pomazánkou a další pekelnosti ZDE',
-            'thumbnail': 're:^http://im.stream.cz/episode/52961d7e19d423f8f06f0100',
-            'duration': 256,
-        },
-    }, {
-        'url': 'http://www.stream.cz/blanik/10002447-tri-roky-pro-mazanka',
-        'md5': '849a88c1e1ca47d41403c2ba5e59e261',
+        # DASH format
+        dash_rel_url = get_url('dash')
+        if dash_rel_url:
+            formats.extend(self._extract_mpd_formats(
+                urljoin(sdn_url, dash_rel_url), video_id, mpd_id='dash',
+                fatal=False
+            ))
+
+        # HLS
+        hls_rel_url = get_url('hls')
+        if hls_rel_url:
+            formats.extend(self._extract_m3u8_formats(
+                urljoin(sdn_url, hls_rel_url), video_id, m3u8_id='hls',
+                ext='ts', fatal=False
+            ))
+
+        self._sort_formats(formats)
+        return formats
+
+
+class StreamCZIE(StreamCZBase):
+    _VALID_URL = r'https?://(?:www\.)?stream\.cz/(?:[^/]+)/.+-(?P<id>[0-9]+)'
+    _TEST = {
+        'url': 'https://www.stream.cz/kdo-to-mluvi/kdo-to-mluvi-velke-odhaleni-prinasi-novy-porad-uz-od-25-srpna-64087937',
         'info_dict': {
-            'id': '10002447',
+            'id': '64087937',
+            'title': 'Kdo to mluví? Velké odhalení přináší nový pořad už od 25. srpna',
+            'description': 'K ikonickým tvářím obrazovek a filmových pláten neodmyslitelně patří i jejich hlasy. Kdo se ale za nimi skrývá? Sledujte nový seriál ze zákulisí dabingu už od příštího úterý.',
+            'timestamp': 1597730400,
+            'upload_date': '20200818',
+            'duration': 51,
             'ext': 'mp4',
-            'title': 'Kancelář Blaník: Tři roky pro Mazánka',
-            'description': 'md5:3862a00ba7bf0b3e44806b544032c859',
-            'thumbnail': 're:^http://im.stream.cz/episode/537f838c50c11f8d21320000',
-            'duration': 368,
-        },
-    }]
+        }
+    }
 
     def _real_extract(self, url):
         video_id = self._match_id(url)
-        api_path = '/episode/%s' % video_id
 
-        req = sanitized_Request(self._API_URL + api_path)
-        req.add_header('Api-Password', _get_api_key(api_path))
-        data = self._download_json(req, video_id)
+        webpage = self._download_webpage(url, video_id)
+        webpage_data = self._html_search_regex(r'window\.APP_SERVER_STATE\s=\s(\{[^;]*)', webpage, 'webpage_data')
```

Comment:
Try `r'(?s)window\s*\.\s*APP_SERVER_STATE\s*=\s*(\{.+?\})\s*;'`:
* `(?s)` to make `.` match across lines (or pass `re.DOTALL` in `flags`)
* all the `\s*` are at points where arbitrary JS whitespace could occur
* `.+?` (1) `.` in case of `;` inside strings in the JSON (2) `+` definitely need some contents (3) `?` don't match `"{...}; ...x = {...};`
* finally force a teminating `}` inside the group and `;` afterwards.